mmv1/third_party/terraform/tests/resource_gkehub_feature_test.go.erb

```diff
@@ -0,0 +1,1092 @@
+package google
+<% unless version == 'ga' -%>
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+  gkehub "github.com/GoogleCloudPlatform/declarative-resource-client-library/services/google/gkehub/beta"
+)
+
+func TestAccGKEHubFeature_gkehubFeatureMciUpdate(t *testing.T) {
+  // VCR fails to handle batched project services
+  skipIfVcr(t)
+  t.Parallel()
+
+  context := map[string]interface{}{
+    "random_suffix":   randString(t, 10),
+    "org_id":          getTestOrgFromEnv(t),
+    "billing_account": getTestBillingAccountFromEnv(t),
+  }
+
+  vcrTest(t, resource.TestCase{
+    PreCheck:  func() { testAccPreCheck(t) },
+    Providers: testAccProvidersOiCS,
+    CheckDestroy: testAccCheckGKEHubFeatureDestroyProducer(t),
+    Steps: []resource.TestStep{
+      {
+        Config: testAccGKEHubFeature_gkehubFeatureMciUpdateStart(context),
+      },
+      {
+        ResourceName:      "google_gkehub_feature.feature",
+        ImportState:       true,
+        ImportStateVerify: true,
+      },
+      {
+        Config: testAccGKEHubFeature_gkehubFeatureMciUpdateEnd(context),
+      },
+      {
+        ResourceName:      "google_gkehub_feature.feature",
+        ImportState:       true,
+        ImportStateVerify: true,
+      },
+    },
+  })
+}
+
+func testAccGKEHubFeature_gkehubFeatureMciUpdateStart(context map[string]interface{}) string {
+  return projectSetup(context) + Nprintf(`
+
+resource "google_container_cluster" "primary" {
+  name               = "tf-test%{random_suffix}"
+  location           = "us-central1-a"
+  initial_node_count = 1
+  project = google_project.project.project_id
+  provider = google-beta
+  depends_on = [google_project_service.mci, google_project_service.container, google_project_service.container, google_project_service.gkehub]
+}
+
+resource "google_container_cluster" "secondary" {
+  name               = "tf-test2%{random_suffix}"
+  location           = "us-central1-a"
+  initial_node_count = 1
+  project = google_project.project.project_id
+  provider = google-beta
+  depends_on = [google_project_service.mci, google_project_service.container, google_project_service.container, google_project_service.gkehub]
+}
+
+resource "google_gke_hub_membership" "membership" {
+  membership_id = "tf-test%{random_suffix}"
+  endpoint {
+    gke_cluster {
+      resource_link = "//container.googleapis.com/${google_container_cluster.primary.id}"
+    }
+  }
+  description = "test resource."
+  project = google_project.project.project_id
+  provider = google-beta
+}
+
+resource "google_gke_hub_membership" "membership_second" {
+  membership_id = "tf-test2%{random_suffix}"
+  endpoint {
+    gke_cluster {
+      resource_link = "//container.googleapis.com/${google_container_cluster.secondary.id}"
+    }
+  }
+  description = "test resource."
+  project = google_project.project.project_id
+  provider = google-beta
+}
+
+resource "google_gkehub_feature" "feature" {
```

Comment:
Can we use an override to make the product names consistent between the membership and feature?

Reply:
I renamed the product to gke_hub to fix this